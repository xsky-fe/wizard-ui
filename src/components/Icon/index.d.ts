import { IconProps } from '../../interface';

export default function Icon(props: IconProps): JSX.Element;