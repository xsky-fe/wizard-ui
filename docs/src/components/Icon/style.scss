.Icon {
  &__ul {
    background-color: #edeef2;
    display: flex;
    flex-wrap: wrap;
    font-size: $font-size-20;
  }
  &__li {
    width: 20%;
    text-align: center;
    padding: 25px 10px;
    &:hover {
      background-color: #dbddec;
    }
    p {
      font-size: $font-size-12;
    }
    .icon {
      font-size: $font-size-20;
      display: block;
      margin-bottom: 10px;
    }
  }
}