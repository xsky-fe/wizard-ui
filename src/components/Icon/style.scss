@import '../../style/variables.scss';
.icon {
  svg {
    display: inline-block;
    stroke-width: 0;
    stroke: currentColor;
    fill: currentColor;
  }
  &.primary {
    color: $purple-normal;
  }
}
