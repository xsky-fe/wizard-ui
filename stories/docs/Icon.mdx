import { Canvas, Meta, ArgTypes } from '@storybook/blocks';

import * as IconStories from '../Icon.stories.tsx';

<Meta of={IconStories} />

# Icon
<em><strong>(wizard-ui自定义组件)</strong></em>
<p>2023-6-19 [fanmengwen](https://github.xsky.com/fan-mengwen)</p>

wizard-ui 内部维护的 [wizard-ui-icons](https://github.xsky.com/front-end/wizard-ui-icon) 图标库, 直观展示某个资源或者某种状态。
## 使用场景


## 代码演示

### 常规
<Canvas of={IconStories.Basic}/>
### 颜色
通过color设置
<Canvas of={IconStories.Color}/>
### 默认继承外部字体大小
默认继承父级字体大小和颜色。
<Canvas of={IconStories.inheritColorSize}/>
### 交互
添加 cursor 属性，实现鼠标 hover 和 active 交互效果，padding 大小为 6px;
<Canvas of={IconStories.Cursor}/>
## 所有图标
<Canvas of={IconStories.allIcons}/>

## API
<ArgTypes />