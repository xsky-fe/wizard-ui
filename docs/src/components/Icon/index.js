import React from 'react';
import * as Icons from 'wizard-ui-icon';
import './style.scss';
import lodash from 'lodash';

const keys = lodash.keys(Icons);

const Icon = () => (
  <div className="Icon">
    <p>总共有 {keys.length} 个图标</p>
    <div className="Icon__ul">
      {keys.map((key, index) => {
        const Icon = Icons[key];
        const name = lodash.kebabCase(key);
        return (
          <div className="Icon__li" key={index}>
            <Icon />
            <p>{name}</p>
          </div>
        );
      })}
    </div>
  </div>
);

export default Icon;
