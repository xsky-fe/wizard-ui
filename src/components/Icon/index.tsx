import * as React from 'react';
import * as PropTypes from 'prop-types';
import { IconProps } from '../../interface';
import icons from 'wizard-icons';
import lodash from 'lodash';
import './style.scss';

const Icon: React.SFC<IconProps> = props => {
  const { type, color, ...rest } = props;
  const cls = `${color} icon icon-${type}`;
  const name = lodash.upperFirst(lodash.camelCase(type));
  const Icon = icons[name];
  return (
    <span className={cls} {...rest} >
      <Icon aria-hidden/>
    </span>
  );

}

Icon.propTypes = {
  /** 图标类型 */
  type: PropTypes.string.isRequired,
  /** 颜色 */
  color: PropTypes.string
}

export default Icon;