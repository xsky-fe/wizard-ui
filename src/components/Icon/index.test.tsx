import React from 'react';
import Icon from './index';
import { render } from 'enzyme';

describe('Loader', () => {
  it('should render default ui', () => {
    const icon = render(<Icon type="os-search-role" />);
    expect(icon).toMatchSnapshot();
  });
});
