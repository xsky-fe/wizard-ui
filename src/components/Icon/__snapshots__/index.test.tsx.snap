// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render default ui 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "undefined icon icon-os-search-role",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "aria-hidden": "true",
          "height": "1em",
          "viewBox": "0 0 34 32",
          "width": "1em",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "d": "M27.337 15.763c-.794-.287-1.65-.443-2.542-.443-4.13 0-7.451 3.34-7.451 7.451 0 3.433 2.318 6.301 5.495 7.148h-18.9c-.854 0-1.547-.692-1.547-1.547V4.465c0-.854.692-1.547 1.547-1.547h21.852c.854 0 1.547.692 1.547 1.547v11.297zm-2.542 1.181c3.2 0 5.827 2.579 5.827 5.779 0 3.248-2.627 5.827-5.827 5.827-3.248 0-5.827-2.579-5.827-5.779s2.579-5.827 5.827-5.827zM5.61 23.679v1.701h8.12v-1.701H5.61zm0-5.413v1.701h11.367v-1.701H5.61zm0-5.413v1.701h14.615v-1.701H5.61zm0-5.413v1.701h16.239V7.44H5.61zm22.003 18.722c.334 0 .621-.287.621-.621 0-.191-.096-.382-.191-.478L26.132 23.2c.239-.382.382-.812.382-1.29 0-1.433-1.146-2.579-2.579-2.579s-2.579 1.146-2.579 2.579a2.568 2.568 0 0 0 2.579 2.579c.478 0 .907-.143 1.29-.382l1.911 1.863c.143.096.287.191.478.191zm-3.677-5.588c.716 0 1.29.573 1.29 1.29s-.573 1.29-1.29 1.29c-.716 0-1.29-.573-1.29-1.29s.573-1.29 1.29-1.29z",
            },
            "children": Array [],
            "name": "path",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "d": undefined,
            },
            "x-attribsPrefix": Object {
              "d": undefined,
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-hidden": undefined,
          "height": undefined,
          "viewBox": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-hidden": undefined,
          "height": undefined,
          "viewBox": undefined,
          "width": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
